webpack.common.js

```diff
@@ -0,0 +1,559 @@
+const autoprefixer = require('autoprefixer');
+const postcssUrl = require('postcss-url');
+const cssnano = require('cssnano');
+const customProperties = require('postcss-custom-properties');
+
+const {NoEmitOnErrorsPlugin, EnvironmentPlugin, HashedModuleIdsPlugin} = require('webpack');
+const {BaseHrefWebpackPlugin, SuppressExtractedTextChunksWebpackPlugin} = require('@angular/cli/plugins/webpack');
+const {CommonsChunkPlugin, ModuleConcatenationPlugin} = require('webpack').optimize;
+const {LicenseWebpackPlugin} = require('license-webpack-plugin');
+const {PurifyPlugin} = require('@angular-devkit/build-optimizer');
+
+const nodeModules = path.join(process.cwd(), 'node_modules');
+const realNodeModules = fs.realpathSync(nodeModules);
+const genDirNodeModules = path.join(process.cwd(), 'src', '$$_gendir', 'node_modules');
+const entryPoints = ["inline", "polyfills", "sw-register", "styles", "vendor", "main"];
+const minimizeCss = true;
+const baseHref = "";
+const deployUrl = "";
+
+const postcssPlugins = function () {
+  // safe settings based on: https://github.com/ben-eb/cssnano/issues/358#issuecomment-283696193
+  const importantCommentRe = /@preserve|@license|[@#]\s*source(?:Mapping)?URL|^!/i;
+  const minimizeOptions = {
+    autoprefixer: false,
+    safe: true,
+    mergeLonghand: false,
+    discardComments: {remove: (comment) => !importantCommentRe.test(comment)}
+  };
+  return [
+    postcssUrl({
+      url: (URL) => {
+        const {url} = URL;
+        // Only convert root relative URLs, which CSS-Loader won't process into require().
+        if (!url.startsWith('/') || url.startsWith('//')) {
+          return URL.url;
+        }
+        if (deployUrl.match(/:\/\//)) {
+          // If deployUrl contains a scheme, ignore baseHref use deployUrl as is.
+          return `${deployUrl.replace(/\/$/, '')}${url}`;
+        }
+        else if (baseHref.match(/:\/\//)) {
+          // If baseHref contains a scheme, include it as is.
+          return baseHref.replace(/\/$/, '') +
+            `/${deployUrl}/${url}`.replace(/\/\/+/g, '/');
+        }
+        else {
+          // Join together base-href, deploy-url and the original URL.
+          // Also dedupe multiple slashes into single ones.
+          return `/${baseHref}/${deployUrl}/${url}`.replace(/\/\/+/g, '/');
+        }
+      }
+    }),
+    autoprefixer(),
+    customProperties({preserve: true})
+  ].concat(minimizeCss ? [cssnano(minimizeOptions)] : []);
+};
+
+module.exports = {
+  "resolve": {
+    "extensions": [
+      ".ts",
+      ".js"
+    ],
+    "modules": [
+      "./node_modules",
+      "./node_modules"
+    ],
+    "symlinks": true,
+    "alias": rxPaths(),
+    "mainFields": [
+      "browser",
+      "module",
+      "main"
+    ]
+  },
+  "resolveLoader": {
+    "modules": [
+      "./node_modules",
+      "./node_modules"
+    ],
+    "alias": rxPaths()
+  },
+  "entry": {
+    "main": [
+      "./src\\main.ts"
```

Comment:
Two backslashes don't seem to be cross-platform compliant. Same goes with many other lines in the code file.